app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/event_catcher/stream.rb

```diff
@@ -21,8 +22,8 @@ def poll
     loop do
       @to = Time.now.to_i
 
-      ems.with_provider_connection(:service => 'events') do |api|
-        events_client = api.sdk_client
+      ems.with_provider_connection do |connect|
+        events_client = connect.events(:region => ems.provider_region, :service_key => @service_key).sdk_client
```

Comment:
Looking for suggestions here, it would be ideal if we didn't have to re-auth for every poll

Reply:
Yeah agreed, better to pull the connection out of the loop like
```ruby
events_client = ems.connect.events(:region => ems.provider_region, :service_key => @service_key).sdk_client
loop do
  ...
end
```

Reply:
@agrare thanks, I've updated with your suggestion